src/libmongoc/src/mongoc/mongoc-topology.c

```diff
@@ -1301,6 +1301,10 @@ mongoc_topology_select_server_id (mongoc_topology_t *topology,
    }
 
 done:
+   if (error) {
+      _mongoc_error_append (error, topology_type->str);
```

Comment:
If `error` is non-NULL and an error was not set, I expect this is appending to an uninitialized `bson_error_t`. This may have been the cause of the Windows task failures.